### DISCUSSION TITLE
setContent and generateJSON remove img nodes when src is a base64 string and not an url. What am I doing wrong?

### DISCUSSION THREAD

**Discussion by tspains** [ueberdosis/tiptap#2498] [General Questions]
Hi, when I try to set the editor's content on initialization with an HTML string, if the string contains a  <img> tag with a base64 encoded source string, setContent removes it. I'm using the Image extension with inline option set to true. Minimal example

`const myContentString = '<p><img src="data:image/png;base64, iVBORw0KGgo..... "/></p>'
editor.commands.setContent(myContentString )`

so I tried to understand what was going on using the generateJSON function from the html extension, and the result is the same. The generated JSON just has the 'doc' and 'paragraph' nodes, not the image.

if instead of a base64 string I use a normal url, it works fine. What am i doing wrong?

Thank you!

**Comment by philippkuehn** [comment]
We don’t provide support for base64 images by default. See: https://github.com/ueberdosis/tiptap/issues/2455

**Reply by tspains** [reply]
Oh ok thank you, I couldn't find this information anywhere.
